sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/TimestampFormatter.scala

```diff
@@ -121,6 +122,10 @@ class Iso8601TimestampFormatter(
   override def parseWithoutTimeZone(s: String): Long = {
```

Comment:
when do we call `parseWithoutTimeZone` today? We need to evaluate other behavior changes we may introduce.

Reply:
It is used to parse any timestamp without timezone, used in timestamp functions and CSV and JSON data sources.

I am a bit confused now. That is why I introduced a separate method isTimeZoneSet to not modify the behaviour of the original method since it did not throw an exception. 

How can one evaluate other behaviour changes?